src/ILCompiler.ReadyToRun/src/IBC/IBCProfileParser.cs

```diff
@@ -0,0 +1,785 @@
+                    TokenList.Count == 0)
+                {
+                    continue;
+                }
+
+                // In V1 and minified V3+ files, tokens aren't stored with a
+                // scenario mask. In the former case, the scenario mask should
+                // be treated as 0 (unprocessed) so that the appropriate flags
+                // will be set. In the latter case, it can be treated as
+                // anything nonzero--minified files make no guarantee about
+                // preserving scenario information, but the flags must be left
+                // alone.
+
+                uint scenarioMaskIfMissing = (parsedData.FormatMajorVersion == 1) ? 0u : 1u;
+
+                foreach (var entry in TokenList)
+                {
+                    //
+                    // Discard any token list entries which refer to the ParamMethodSpec blob stream entries
+                    // (if any) which were thrown away above.  Note that the MethodProfilingData token list is
+                    // the only place anywhere in the IBC data which can ever contain an embedded ibcMethodSpec
+                    // token.
+                    //
+
+                    if (section == SectionFormat.MethodProfilingData)
+                    {
+                        if (ignoredIbcMethodSpecTokens.Contains(entry.Token))
+                            continue;
+                    }
+
+                    uint scenarioMask = entry.ScenarioMask ?? scenarioMaskIfMissing;
+
+                    // scenarioMask will be 0 in unprocessed or V1 IBC data.
+                    if (scenarioMask == 0)
+                    {
+                        throw new NotImplementedException();
+                        /*                        Debug.Assert(fullScenarioMask == 1, "Token entry not owned by one scenario");
+                                                // We have to compute the RunOnceMethod and RunNeverMethod flags.
+                                                entry.Flags = result.GetFlags(entry.Flags, section, entry.Token);
+                                                scenarioMask = defaultScenarioMask;*/
+                    }
+
+                    //                    Debug.Assert(((~fullScenarioMask & scenarioMask) == 0), "Illegal scenarios mask");
+
+                    MethodDesc associatedMethod = null;
+
+                    switch (Cor.Macros.TypeFromToken(entry.Token))
+                    {
+                        case CorTokenType.mdtMethodDef:
+                        case CorTokenType.mdtMemberRef:
+                        case CorTokenType.mdtMethodSpec:
+                            associatedMethod = ecmaModule.GetMethod(System.Reflection.Metadata.Ecma335.MetadataTokens.EntityHandle((int)entry.Token));
+                            break;
+
+                        /*                        case CorTokenType.ibcExternalMethod:
```

Comment:
Why is this commented out?

Reply:
It turns out that an ExternalMethod can't actually be top level. Its just junk code. I meant to delete it.